pulsar-client-cpp/include/pulsar/MessageId.h

```diff
@@ -34,13 +35,37 @@ class MessageId {
  public:
     MessageId& operator=(const MessageId&);
     MessageId();
+    virtual ~MessageId() {}
+
+    /**
+     * MessageId representing the "earliest" or "oldest available" message stored in the topic
+     */
+    static const MessageId& earliest();
+
+    /**
+     * MessageId representing the "latest" or "last published" message in the topic
+     */
+    static const MessageId& latest();
+
+    /**
+     * Serialize the message id into a binary string for storing
+     */
+    virtual void serialize(std::string& result) const;
```

Comment:
When would we want to use the serialize and deserialize API??

Reply:
When using reader, you need to be able to store the message id somewhere (since you don't have a durable cursor, like in case of subscriptions/consumers). 

Then you can deserialize that message id and use it to create a new Reader starting from a particular message.